From 5612509c21f723bfeae901a1801f64c57545bd25 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:03:16 -0400
Subject: [fs] btrfs: Remove unnecessary check for NULL

Message-id: <1407179109-13714-310-git-send-email-sandeen@redhat.com>
Patchwork-id: 87272
O-Subject: [RHEL7.1 PATCH 309/422] btrfs: Remove unnecessary check for NULL
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Tobias Klauser <tklauser@distanz.ch>

commit 7fb18a06644a33f0a2eba810d57d4b3b2c982f5b
Author: Tobias Klauser <tklauser@distanz.ch>
Date:   Fri Apr 25 14:58:05 2014 +0200

    btrfs: Remove unnecessary check for NULL

    iput() already checks for the inode being NULL, thus it's unnecessary to
    check before calling.

    Signed-off-by: Tobias Klauser <tklauser@distanz.ch>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/scrub.c b/fs/btrfs/scrub.c
index 68a5a26..bd7a4af 100644
--- a/fs/btrfs/scrub.c
+++ b/fs/btrfs/scrub.c
@@ -717,8 +717,8 @@ static int scrub_fixup_readpage(u64 inum, u64 offset, u64 root, void *fixup_ctx)
 out:
  if (page)
   put_page(page);
- if (inode)
-  iput(inode);
+
+ iput(inode);
 
  if (ret < 0)
   return ret;
-- 
1.7.1